### PR TITLE
Various quality of life improvements

### PR DESCRIPTION
**Pull request by fiskenslakt**
Added:
* New commands
  * /rename (update `name` column of a user)
  * /endow (makes user a point giver)
  * /disendow (removes ability to give points from user)
* `point_giver` column
  * determines user's ability to give points
  * `False` by default
* Can set `point_giver` when registering new user
* Helper function to validate names

Changed:
* Timestamps will use special discord timestamp syntax going forward so they are in the timezone of the discord user viewing it
* pycord version bump
* special "everyone" syntax requires both being a discord user and being a point giver to receive points
* /user command has been updated considerably
  * Shows all users instead of just showing ones that are active
  * Paginated, sorted, and in a single column
  * Shows more data about each user
* Point syntax no longer supports `for <recipient>`
* Moved various points related settings to settings.py

### CODE REVIEW

**Comment by rbrammen** [comment]
Approved for PROD production push

File: pzsd_bot/cogs/points.py [hunk 1]
```diff
@@ -53,6 +60,13 @@ def page_buttons(self) -> list[PaginatorButton]:
             PaginatorButton("last", label=">>", style=discord.ButtonStyle.blurple),
         ]
 
+    def validate_name(self, name: str) -> Enum:
+        if re.fullmatch(r"(?:every|no)[ -]?(?:one|body)|me", name):
```
Comment (deforestg) [inline pzsd_bot/cogs/points.py:64]: are all of these possibilities really all reserved names? "every body" "no one" "no-one" etc? I would think there should be a discrete set of reserved names that are listed out

Comment (fiskenslakt) [reply pzsd_bot/cogs/points.py:64]: That's the idea. I suppose I could switch to discretely listing out every name and all its variations, but I just felt like the actual amount of names I don't want people to use was pretty small, and their variations rather large.

Comment (deforestg) [reply pzsd_bot/cogs/points.py:64]: why are all these permutations even reserved? I only see this one hard coded reference here https://github.com/fiskenslakt/pzsd-bot/blob/7818e5f71f3657b57f9041415196e76cabb5475c/pzsd_bot/cogs/points.py#L104

Comment (deforestg) [reply pzsd_bot/cogs/points.py:64]: actually there are a few hard coded references to "everyone" but none of the other permutations

Comment (deforestg) [reply pzsd_bot/cogs/points.py:64]: you should just make "everyone" into a constant somewhere and reference that

File: pzsd_bot/cogs/points.py [hunk 2]
```diff
@@ -98,6 +111,22 @@ async def on_message(self, message: Message) -> None:
                 )
                 return
 
+            if not bestower.is_active:
+                logger.info(
```
Comment (sethdandridge) [inline pzsd_bot/cogs/points.py:115]: i'd make this a warning

File: pzsd_bot/cogs/points.py [hunk 3]
```diff
@@ -263,7 +299,7 @@ async def weekly(self, ctx: ApplicationContext) -> None:
 
         last_week = datetime.now() - timedelta(days=7)
         leaderboard = await self.fetch_leaderboard(ledger.c.created_at > last_week)
-        description = "Points awarded after " + last_week.strftime("%a %b %-d %-I:%M%p")
+        description = "Points awarded after " + f"<t:{int(last_week.timestamp())}:f>"
```
Comment (sethdandridge) [inline pzsd_bot/cogs/points.py:302]:
```suggestion
        description = f"Points awarded after <t:{int(last_week.timestamp())}:f>"
```

File: pzsd_bot/cogs/points.py [hunk 4]
```diff
@@ -404,19 +454,148 @@ async def unregister(self, ctx: ApplicationContext, name: str) -> None:
...
+        async with Session.begin() as session:
+            await session.execute(
+                update(pzsd_user)
+                .where(pzsd_user.c.id == user_to_disendow.id)
+                .values(point_giver=False)
+            )
 
-        await ctx.respond(embed=embed)
+        logger.info("Removed ability to give points from user '%s'", user)
+        await ctx.respond(f"Disendowed {user}")
```
Comment (sethdandridge) [inline pzsd_bot/cogs/points.py:598]: 600+ line file kind of sus

Comment (fiskenslakt) [reply pzsd_bot/cogs/points.py:598]: It's getting a little chonky yea, I haven't really come up with any good ideas on how to split it up yet though.

Comment (fiskenslakt) [reply pzsd_bot/cogs/points.py:602]: Will resolve in a separate PR. I've created #41 for now.

**Review by deforestg — APPROVED** [review]

File: pzsd_bot/cogs/points.py [hunk 5]
```diff
@@ -53,18 +56,25 @@ def page_buttons(self) -> list[PaginatorButton]:
             PaginatorButton("last", label=">>", style=discord.ButtonStyle.blurple),
         ]
 
+    def validate_name(self, name: str) -> Enum:
```
Comment (sethdandridge) [inline pzsd_bot/cogs/points.py:59]: Return hint should be NameState

Comment (fiskenslakt) [reply pzsd_bot/cogs/points.py:59]: Oh yea, I suppose.

File: pzsd_bot/settings.py [hunk 6]
```diff
@@ -41,5 +42,48 @@ class _DB(EnvConfig):
 
 DB_CONNECTION_STR = f"postgresql+asyncpg://{DB.pguser}:{DB.pgpassword}@{DB.pghost}:{DB.pgport}/{DB.pgdatabase}"
 
+
+class _PointsSettings(EnvConfig):
```
Comment (sethdandridge) [inline pzsd_bot/settings.py:46]: Badly named

Comment (sethdandridge) [reply pzsd_bot/settings.py:46]: ParsingSettings

Comment (sethdandridge) [reply pzsd_bot/settings.py:46]: ParsingConstants

Comment (fiskenslakt) [reply pzsd_bot/settings.py:46]: I didn't love it either, but I couldn't name it `_Points` like I originally wanted to without renaming the `Points` class in points.py. And I can't think of an alternate name. Suggestions?

Comment (sethdandridge) [reply pzsd_bot/settings.py:46]: ParsingConstants

**Review by sethdandridge — APPROVED** [review]
lgtmpc

File: pzsd_bot/settings.py (hunk 6 above)
Comment (fiskenslakt) [reply pzsd_bot/settings.py]: Not a fan of these names lol. I'm not opposed to renaming in the future, but until we come up with a better one, I'm keeping it the same for now.

**Review by sethdandridge — CHANGES_REQUESTED** [review]
please change name of PointsSettings. This is a critical blocker

**Comment by fiskenslakt** [comment]
> please change name of PointsSettings. This is a critical blocker

# OVERRULED

**Pull request merged**